mos-platform/common/c/string.c

```diff
@@ -63,3 +63,70 @@ __attribute__((weak)) char *strrchr(const char *s, int c) {
       last = s;
   return (char *)last;
 }
+
+__attribute__((weak)) char* strrev(char* str)
+{
+    size_t len = strlen((const char*)str);
+    size_t i, j;
+    for (i = 0, j = len - 1; i < j; i++, j--)
```

Comment:
nit: prefer declaring variables inside loops if only used inside, here and throughout. The llvm-mos stdlib generally targets c99, and even C++ is allowed if ABI-compatible.

Reply:
I actually had to look up how to declare two variables at once in a for loop. I think I died inside a little. Done.